sdk/samples/iot/paho_iot_hub_twin_sample.c

```diff
@@ -407,75 +403,88 @@ static void parse_device_twin_message(
     case AZ_IOT_CLIENT_TWIN_RESPONSE_TYPE_DESIRED_PROPERTIES:
       LOG("Message Type: Desired Properties");
 
-      if (az_failed(rc = update_property(message_span)))
+      if (!update_property(message_span))
       {
-        LOG_ERROR("Failed to update property locally: az_result return code 0x%08x.", rc);
-        exit(rc);
+        LOG_ERROR("Failed to update property locally.");
+        exit(1);
       }
       break;
   }
 }
 
-static az_result build_reported_property(az_span destination, az_span* json_out)
+static bool build_reported_property(az_span destination, az_span* json_out)
 {
   az_json_writer json_writer;
 
-  AZ_RETURN_IF_FAILED(az_json_writer_init(&json_writer, destination, NULL));
-  AZ_RETURN_IF_FAILED(az_json_writer_append_begin_object(&json_writer));
-  AZ_RETURN_IF_FAILED(az_json_writer_append_property_name(&json_writer, reported_property_name));
-  AZ_RETURN_IF_FAILED(az_json_writer_append_int32(&json_writer, reported_property_value));
-  AZ_RETURN_IF_FAILED(az_json_writer_append_end_object(&json_writer));
+  if (az_failed(az_json_writer_init(&json_writer, destination, NULL))
+      || az_failed(az_json_writer_append_begin_object(&json_writer))
+      || az_failed(az_json_writer_append_property_name(&json_writer, reported_property_name))
+      || az_failed(az_json_writer_append_int32(&json_writer, reported_property_value))
+      || az_failed(az_json_writer_append_end_object(&json_writer)))
+  {
+    return false;
+  }
```

Comment:
> ... and returning az_sresult by customer functions (IoT samples)

Why did you want to do this? Don't we hurt some diagnosability by not printing the reason for the failure, which `az_result` was built for? The type is public.